tests/test_recursive.py

```diff
@@ -72,21 +70,48 @@ def member(self, model):
 
 class SubMachine(Machine2):
     def __init__(self):
-        _super(SubMachine, self).__init__()
-        #super(SubMachine, self).__init__() #XXX: works, except for 3.1-3.3
+        super(SubMachine, self).__init__()
 
 
 def test_partials():
     assert dill.copy(SubMachine(), byref=True)
     assert dill.copy(SubMachine(), byref=True, recurse=True)
-   #if not OLDER: #FIXME: fails __main__.py
-   #    assert dill.copy(SubMachine(), recurse=True)
+    assert dill.copy(SubMachine(), recurse=True)
     assert dill.copy(SubMachine())
 
 
+class obj4(object):
+    def __init__(self):
+        super(obj4, self).__init__()
+        a = self
+        class obj5(object):
+            def __init__(self):
+                super(obj5, self).__init__()
+                self.a = a
+        self.b = obj5()
+
+
+def test_circular_reference():
+    assert dill.copy(obj4())
+    obj4_copy = dill.loads(dill.dumps(obj4()))
+    if PY3:
```

Comment:
Just to confirm, this is a PY3-only use case, correct?

Reply:
Because the system behind `super` is different in Python 2 and Python 3. In Python 2, cells are not used to represent `super`, so the checking of the contents of the cell is only relevant in Python 3.